veps/sig-compute/decentralized-migration/decentralized-migration-vep.md

```diff
@@ -0,0 +1,279 @@
+ - Create a synchronization flow to allow the synchronization of two `VirtualMachineInstance` statuses.
+ - Update the `VirtualMachineInstance` migrationState to support those separate entities and to allow the virt-controller, and virt-handler to identify which entity they are processing.
+
+## Non Goals
+
+Orchestration of the building blocks provided by this proposal. A higher level orchestrator is required to make usage easier for end users.
+
+Ensuring connectivity between clusters if live migrating between clusters.
+
+## Definition of Users
+
+- Cluster Admins
+- Admins managing several clusters
+
+## User Stories
+
+- As a KubeVirt admin I would like to evacuate my cluster so I can do disruptive maintance on this cluster (ex. hardware refresh) without disrupting the workloads.
+- As a KubeVirt admin maintaining several kubernetes clusters, I would like to rebalance the virtual machines running on my clusters to even out the load of my clusters. without disrupting the workloads.
+- As a KubeVirt namespace admin, I would like to move a virtual machine from one namespace to another without disrupting the workloads.
+
+## Repos
+
+[KubeVirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
+#### Current state
+To start a live migration one creates a `VirtualMachineInstanceMigration` resource in the cluster, which references the `VirtualMachineInstance` you want to migrate. This then creates a _receiver_ pod in the same namespace, and starts the migration process.
+
+The actual migration channel is proxied through the `virt-handler` pods that are running on the source and target nodes. Any migration state information is passed through the `VirtualMachineInstance` migrationStatus. This is because `virt-handler` pods can only read/modify `VirtualMachineInstances`.
+
+The reason virt-launcher pods do not communicate directly with each other, is that it is not guaranteed that virt-launchers will have network connectivity. The virt-handler pods also have access to a migration network if one is defined.
+
+The migration channel in figure 1 is either the pod network, or a specially configured migration network.
+
+Figure 1 describes the current interaction between various components:
+![figure 1](migration_current_state.png)
+
+#### Proposed state
+To start a live migration one creates two `VirtualMachineInstanceMigration` resources. A `target` resource, and a `source` resource.
+
+The target cluster will have a synchronization controller running in the kubevirt namespace, that has access to the migration channel (either pod network, or migration network). This controller waits for incoming synchronization connections from a source. When a `source` migration resource is created in the cluster, this synchronization controller will connect to the target synchronization and start synchronizing the `VirtualMachineInstance` status between them.
```

Comment:
What exactly will the synchronizatoin controller do?

In my limited understanding I'd assume we launch
dst: libvirt listening
src: libvirt sending to dst

I'm missing what the ctrler does.

Reply:
The synchronization controller makes it so we don't have to change the logic of how migrations work. Right now as explained in this document, we pass information between the source and target through the VMI. But with this proposal we end up with two different VMIs, so we can't just directly write to a VMI and expect the other side to see it. Or can we? Well that is the purpose of the synchronization controller. To synchronize the VMIs so we can pass information back and forth like we do directly with the VMI now.

Reply:
I added some verbage earlier in the proposal to explain the role of the synchronization controller, before we get to the synchronization section.